src/prommis/nanofiltration/diafiltration_two_salt.py

```diff
@@ -0,0 +1,1116 @@
+**useDefault** - use default package from parent model or flowsheet,
+**PhysicalParameterObject** - a PhysicalParameterBlock object.}
+""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+and used when constructing these,
+**default** - None.
+**Valid values:** {see property package for documentation}
+""",
+        ),
+    )
+    CONFIG.declare(
+        "NFEx",
+        ConfigValue(
+            doc="Number of discretization points in the x-direction",
+        ),
+    )
+    CONFIG.declare(
+        "NFEz",
+        ConfigValue(
+            doc="Number of discretization points in the z-direction",
+        ),
+    )
+
+    def build(self):
+        """
+        Build method for the two salt diafiltration unit model
+        """
+        super().build()
+
+        # TODO: generalize to any 2 cations and 1 anion
+
+        self.add_mutable_parameters()
+        self.add_variables()
+        self.add_constraints()
+        self.discretize_model()
+        self.fix_initial_values()
+        self.add_scaling_factors()
+        self.add_ports()
+
+    def add_mutable_parameters(self):
+        """
+        Adds parameters for the two salt diafiltration unit model.
+
+        Assigns default values that can be changed by the user during implementation.
+        """
+        self.numerical_zero_tolerance = Param(
+            initialize=1e-10,
+            mutable=True,
+            doc="Numerical tolerance for zero values in the model",
+        )
+        self.membrane_thickness = Param(
+            initialize=1e-7,
+            mutable=True,
+            units=units.m,
+            doc="Thickness of membrane (z-direction)",
+        )
+        self.membrane_permeability = Param(
+            initialize=0.03,
+            mutable=True,
+            units=units.m / units.h / units.bar,
+            doc="Hydraulic permeability coefficient",
+        )
+        self.temperature = Param(
+            initialize=298,
+            mutable=True,
+            units=units.K,
+            doc="System temperature",
+        )
+
+    def add_variables(self):
+        """
+        Adds variables for the two salt diafiltration unit model.
+        """
+        # define length scales
+        self.x_bar = ContinuousSet(bounds=(0, 1))
+        self.z_bar = ContinuousSet(bounds=(0, 1))
+
+        # add a time index since the property package variables are indexed over time
+        self.time = Set(initialize=[0])
+
+        # add components
+        self.solutes = Set(initialize=["Li", "Co", "Cl"])
+
+        # add global variables
+        self.membrane_width = Var(
+            initialize=1,
+            units=units.m,
+            domain=NonNegativeReals,
+            doc="Width of the membrane (x-direction)",
+        )
```

Comment:
Generally we prefer to set `bounds=[0, None]` instead of a domain. There are situations where we'd like to get rid of the lower bound of a `Var` in order to avoid issues with IPOPT, but `x.setlb(None)` isn't sufficient. First you need to override the domain, _then_ modify the lower bound. `x.setlb(None)` silently fails, and you'll find `x.bounds == [0, None]` if you don't correct the domain first.